File: api/src/Retroscope.Application/DTOs/FeatureFlags.cs
namespace Retroscope.Application.DTOs;

public class FeatureFlagItem
{
    public string FlagName { get; set; } = string.Empty;
    public string? Description { get; set; }
    public bool IsEnabled { get; set; }
}

public class FeatureFlagsResponse
{
    public List<FeatureFlagItem> Items { get; set; } = new();
}

File: src/lib/data/csharpApi/featureFlags.ts
import { fetchApi } from '@/lib/data/csharpApi/utils';

export async function apiGetFeatureFlags(): Promise<{ items: Array<{ flagName: string; description?: string | null; isEnabled: boolean }> }> {
    const res = await fetchApi(`/api/featureflags`);
    return res.json();
}

export async function apiUpdateFeatureFlag(flagName: string, isEnabled: boolean): Promise<void> {
    const res = await fetchApi(`/api/featureflags/${encodeURIComponent(flagName)}`, {
        method: 'PATCH',
        body: JSON.stringify(isEnabled)
    });
    return res.json();
}